src/airflow_tools/providers/filesystem/operators/filesystem.py

```diff
@@ -55,7 +102,12 @@ def execute(self, context):
             connection=BaseHook.get_connection(self.destination_fs_conn_id),
         )
 
-        for file_path in source_fs_hook.list_files(self.source_path):
+        files = (
+            source_fs_hook.list_files(self.source_path)
+            if source_fs_hook.check_prefix(self.source_path)
```

Comment:
Does this work for object storages like S3 and blob storage or unity catalog volume? I'm not sure if they count a full object path as a prefix